splunk-adx-alert-addon/adx_connector.py

```diff
@@ -0,0 +1,205 @@
+                    except Exception as err:
+                        #if any error, continue, the file will be picked up in the next event invocation
+                        helper.log_error("Error while moving file to temp folder {}".format(err))
+                        continue
+                else:
+                     helper.log_error("File Ingestion failed, will retry ingestion in the next event invocation = {}".format(file_path))
+                     continue
+    
+
+    def _get_files_not_modified(self,directory):
+        current_time = time.time()
+        files = []
+        for file_name in os.listdir(directory):
+            file_path = os.path.join(directory, file_name)
+            # Check if it is a regular file
+            if os.path.isfile(file_path):
+                # Get the file's modification time
+                modification_time = os.path.getmtime(file_path)
+                # Compare the modification time with the current time
+                if modification_time < current_time:
+                    files.append(file_path)
+        return files
+    
+
+    def _move_files_to_temp_dir(self,helper,source_file_path):
+        #create temp dir if not exists
+        if not os.path.isdir(os.path.join(tempfile.gettempdir(), "ADX_Ingest")):
+            temp_dir_created = tempfile.mkdtemp()
+            temp_dir = os.path.join(os.path.dirname(temp_dir_created), "ADX_Ingest")
+            os.rename(temp_dir_created, temp_dir)
+        else:
+            temp_dir = os.path.join(tempfile.gettempdir(), "ADX_Ingest")
+        if os.path.isfile(os.path.join(temp_dir, os.path.basename(source_file_path))):
+            new_file_name = self._get_new_file_name(os.path.basename(source_file_path))
+            shutil.move(source_file_path, os.path.join(temp_dir, new_file_name))
+        else:
+            shutil.move(source_file_path, temp_dir)
+
+    def _get_new_file_name(self,file_name):
+        base_name, ext = os.path.splitext(file_name)
+        counter = 1
+        new_file_name = f"{base_name}_{counter}{ext}"
+        while os.path.exists(os.path.join(new_file_name)):
+            counter += 1
+            new_file_name = f"{base_name}_{counter}{ext}"
+        return new_file_name    
+
+    def _is_file_not_being_modified(self,file_path):
+        """
+        Returns True if the file is currently being not modified, False otherwise.
+        """
+        if os.path.getsize(file_path) > max_size:
+            return True
+        else : 
+            file_stat = os.stat(file_path)
+            current_size = file_stat.st_size
+            time.sleep(10)  # Wait for 5 second
+            new_size = os.stat(file_path).st_size
+            # Get the file's modification time
+            modification_time = os.path.getmtime(file_path)
+            # Compare the modification time with the current time
+            current_time = time.time()
```

Comment:
Do we have to take care of explicitly calling out UTC ? (or if all are in the same unit it is fine, just food for thought)